docs/content/concepts/transactions/WIP-ptb.md

```diff
@@ -0,0 +1,400 @@
+  - The gas coin cannot be taken by-value except with the `TransferObjects` command. If you need an owned version of the gas coin, you can first use `SplitCoins` to split off a single coin.
+  - This limitation exists to make it easy for the remaining gas to be returned to the coin at the beginning of execution. In other words, if the gas coin was wrapped or deleted, then there would not be an obvious spot for the excess gas to be returned. See the execution section for more details.
+- `NestedResult(u16, u16)` uses the value from a previous command. The first `u16` is the index of the command in the command vector, and the second `u16` is the index of the result in the result vector of that command.
+  - For example, given a command vector of `[MoveCall1, MoveCall2, TransferObjects]` where `MoveCall2` has a result vector of `[Value1, Value2]`, `Value1` would be accessed with `NestedResult(1, 0)` and `Value2` would be accessed with `NestedResult(1, 1)`.
+- `Result(u16)` is a special form of `NestedResult` where `Result(i)` is equivalent to `NestedResult(i, 0)`. However, this will error if the result array at index `i` is empty or has more than one value.
+  - The intention of `Result` was to allow for accessing of the entire result array, but that is not yet supported at this time. So in general, `NestedResult` should be used instead of `Result`.
+
+## Execution
+
+For the execution of programmable transaxtion block: the input vector is populated by the input objects or pure value bytes. Then the commands are executed in order, and the results are stored in the result vector. Finally, the effects of the transaction are applied atomically. The following sections will describe each aspect of execution in greater detail.
+
+### Start of Execution
+
+At the beginning of execution, the programmable transaction block runtime takes the already loaded input objects and loads them into the input array. The objects are already verified by the network, checking rules like existence and valid ownership. The pure value bytes are also loaded into the array but not validated until usage.
+
+The most important thing to note at this stage is the effects on the gas coin. At the beginning of execution, the maximum gas budget (in terms of `SUI`) is withdrawn from the gas coin. Any unused gas will be returned to the gas coin at the end of execution, even if the coin has changed owners!
+
+### Executing a Command
+
+Each command is then executed in order. First, let's look a the rules around arguments, which are shared by all commands.
+
+#### Arguments
+
+- Each argument can be used by-reference or by-value. The usage is based on the type of the argument and the type signature of the command.
+  - If the signature expects an `&mut T`, the runtime checks the argument has type `T` and it is then mutably borrowed.
```

Comment:
It feels like a lot to digest before getting to the description of commands, and I am not sure all of this needs to be digested before reading description of commands. Should we perhaps consider moving this to some kind of "additional details" section (at the end of "Executing a Command")?

Reply:
This document is supposed to be more of a reference, so I am fine leaning a bit heavier here. I had intended to put in all the detail here because it allows me to accurately describe each command more succinctly, though I totally get what you are saying. 

Maybe @ronny-mysten will have ideas?

Reply:
I'm of the same mind as Todd. I don't think it's an issue because I don't see it as a doc someone will read from beginning to end at once. We can always revisit based on user feedback.